### DISCUSSION TITLE
Downloads to 64bit RAM address space?

### DISCUSSION THREAD

**Discussion by d-brtn** [ipxe/ipxe#1080] [Ideas]
From what I can tell downloads are limited to 32bit memory address space with things like imgfetch. Does someone with a little bit of more knowledge of the internals of the project, know if this is something that could be modified by an end user/easily patched in source? 
I'm working in circumstance where it is very desirable to do all networking before any OS is loaded, and this limits payload sizes on machines with absurdly large quantities of RAM.

**Comment by NiKiZe** [comment]
What are you basing the assumption of 32bit limit on?
What platform are you booting (pcbios or efi)?
